econml/cate_estimator.py

```diff
@@ -660,6 +751,52 @@ def summary(self, alpha=0.1, value=0, decimals=3, feat_name=None, treatment_name
         if len(smry.tables) > 0:
             return smry
 
+    def shap_values(self, X, *, feature_names=None, treatment_names=None, output_names=None):
+        d_t = self._d_t[0] if self._d_t else 1
+        d_y = self._d_y[0] if self._d_y else 1
+
+        if treatment_names is None:
+            treatment_names = [f"T{i}" for i in range(d_t)]
+        if output_names is None:
+            output_names = [f"Y{i}" for i in range(d_y)]
+
+        Xs, T = broadcast_unit_treatments(X, d_t)
+        if self.featurizer is not None:
+            F = self.featurizer.transform(Xs)
+        else:
+            F = Xs
+        d_x = F.shape[1]
+        X_new = cross_product(F, T)
+        feature_names = self.cate_feature_names(feature_names)
+        if self.fit_cate_intercept and feature_names is not None:
+            feature_names = ["Intercept"] + feature_names
+        if feature_names is not None:
+            feature_names = np.tile(feature_names, d_t)
+
+        shap_outs = defaultdict(dict)
+        for i in range(d_t):
+            X_sub = X_new[T[:, i] == 1]
+            ind_x = list(range(i * d_x, (i + 1) * d_x)
+                         )[1:] if self.fit_cate_intercept else list(range(i * d_x, (i + 1) * d_x))
```

Comment:
But actually it might be cleaner to define ind_x as a 2D array outside the loop:

```
ind_x = np.arange(d_t * d_x).reshape(d_t, d_x)
if self.fit_cate_intercept:
    ind_x = ind_x[:,1:]
```

and then inside the loop just take the `i`th row

Reply:
Great! Done!